includes/subscriptions/class-wc-payments-subscription-service.php

```diff
@@ -930,4 +932,51 @@ private function get_item_metadata( WC_Order_Item $item ) {
 
 		return $metadata;
 	}
+
+	/**
+	 * Validates that the data used to create the WCPay Subscription.
+	 *
+	 * @param array $subscription_data The data used to create a WCPay subscription.
+	 * @throws Exception If the subscription data contains invalid or missing data.
+	 */
+	private function validate_subscription_data( $subscription_data ) {
+
+		if ( empty( $subscription_data['customer'] ) ) {
+			throw new Exception( 'The "customer" arg is required to create the subscription.' );
+		}
+
+		if ( ! isset( $subscription_data['items'] ) ) {
+			throw new Exception( 'The "items" arg is required to create the subscription.' );
+		}
+
+		foreach ( $subscription_data['items'] as $item_data ) {
+			$required_price_keys  = [ 'currency', 'product', 'unit_amount_decimal', 'recurring' ];
+			$required_period_keys = [ 'interval', 'interval_count' ];
+			$errors               = [];
+
+			foreach ( $required_price_keys as $required_key ) {
+				if ( empty( $item_data['price_data'][ $required_key ] ) ) {
```

Comment:
I think this will cause problems with a $0 subscription that have other costs associated with it.

**Steps to reproduce**
- Set up flat shipping to all countries (i.e. $10)
- Create a $0 subscription.
- As a shopper, add the product to the cart and navigate to checkout.
- Observe the subscription cost is $0 and shipping is $10.
- Attempt to purchase the subscription via WCPay card payment method.
- Observe the checkout error "There was a problem creating your subscription. Please try again or contact us for assistance."
- Observe the logs:

```
INFO There was a problem creating the WCPay subscription. The "unit_amount_decimal" line item properties are required to create the subscription.
```

I believe this would also happen for $0 subscriptions with a sign-up fee.

Could we use `! isset()` for `unit_amount_decimal'` and `empty()` for the other keys?

Reply:
😱 Oh good catch. I've replicated that and patched it with 498047e.